### ISSUE TITLE
Feature Request: Extending `@typescript-eslint/no-magic-numbers` to allow for `readonly number[]`

### ISSUE THREAD

**Issue by Samuel-Therrien-Beslogic** [eslint-stylistic/eslint-stylistic#238]
### Clear and concise description of the problem

There's a use case that comes up once in a while to define "magic" numbers in an array that itself is meant to be used as a constant.

### Suggested solution

Copied from https://github.com/typescript-eslint/typescript-eslint/issues/8102 . Thoughts on allowing (can be locked behind an option) the following?

### Fail

```TypeScript
const DEFAULT_PAGE_LENGTH = [5, 10, 25, 100]

class MyCLass {
  readonly DEFAULT_PAGE_LENGTH = [5, 10, 25, 100]
}

let DEFAULT_PAGE_LENGTH = [5, 10, 25, 100] as const

class MyCLass {
  DEFAULT_PAGE_LENGTH = [5, 10, 25, 100] as const
}
```

### Pass

```TypeScript
const DEFAULT_PAGE_LENGTH = [5, 10, 25, 100] as const

class MyCLass {
  readonly DEFAULT_PAGE_LENGTH = [5, 10, 25, 100] as const
}
```

### Alternative

The alternative to a eslint-disable comment really isn't preferable and really verbose:
```ts
const STEP_1 = 5
const STEP_2 = 10
const STEP_3 = 25
const STEP_4 = 10
const DEFAULT_PAGE_LENGTH = [STEP_1 , STEP_2 , STEP_3 , STEP_4]
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by antfu** [comment]
Well, `no-magic-number` does not seem to be part of this project.

**Comment by antfu** [comment]
As stated in https://github.com/eslint-stylistic/eslint-stylistic/issues/26#issuecomment-1773890620, we don't have a plan to migrate rules if eslint/ts-eslint teams are not deprecating them. Close if for now.
